api/src/main/java/org/openmrs/hl7/handler/ORUR01Handler.java

```diff
@@ -355,12 +355,13 @@ private Message processORU_R01(ORU_R01 oru) throws HL7Exception {
 				}
 				catch (HL7Exception e) {
 					errorInHL7Queue = e;
-				}
+					throw new HL7Exception e(Context.getMessageSourceService().getMessage(
+						    "ORUR01.error.improperlyFormattedOBX";
+					}
 				finally {
 					// Handle obs-level exceptions
-					if (errorInHL7Queue != null) {
-						throw new HL7Exception(Context.getMessageSourceService().getMessage(
-						    "ORUR01.error.improperlyFormattedOBX",
+					if (errorInHL7Queue != null) 
```

Comment:
I think you forgot to put the braces after the ```if``` statement on ```L363```

Reply:
I am removing this if block as suggested by @PrathameshMutkure

Reply:
Don't just remove but also put it somewhere else outside finally block

Reply:
are you talking about braces